### DISCUSSION TITLE
Personal API Keys - Self Generation

### DISCUSSION THREAD

**Discussion by cygars** [etsy/open-api#1194] [Q&A]
Greetings all,

why is this not a feature in the developer portal, that you can self generate api keys for personal use and regenerate them on demand but you have to approve them?

Kind regards.

**Comment by mifloma** [comment]
I agree. It doesn't makes sense to me that I have Etsy to approve access to my own shop data. Especially for creating POCs, quick personal access would be very helpful.

**Comment by constantin-bogdan** [comment]
Hello!

I got my first app request rejected, because I named the app "EtsyOrderImporter" and per TOS it's not allowed to use "Etsy" in the name of your personal app. Ok, questionable, but ok. Now I have requested a new App with a more neutral name and I'm waiting for 20 days for approval. 
This is not acceptable in my opinnion. 

King regards.

**Comment by schang969** [comment]
I also think it's counter-productive for Etsy to deny access to my own shop's data. Especially after I learned Python only to automate my own transactions. 

Does Etsy have someone patrolling this forum though?
